src/utils/hooks/naiveBranchingDialogWizardHook.ts

```diff
@@ -0,0 +1,90 @@
+import { ComponentType } from 'react';
+
+import { useDialog, DialogType } from '~core/Dialog';
+import { isDev } from '~utils/debug';
+
+type PotentialProps = Record<string, any>;
+
+interface Step {
+  component: ComponentType<any>;
+  props?: PotentialProps;
+}
+
+type Steps = Step[];
+
+type CallStepFnSignature = (dialogName: string) => void;
+
+export type WizardDialogType<P> = DialogType<P | undefined> & {
+  callStep: CallStepFnSignature;
+};
+
+interface WizardComponent {
+  component: <P extends object>(potentialProps?: P) => WizardDialogType<P>;
+  props?: PotentialProps;
+}
+
+type WizardMap = Record<string, WizardComponent>;
+
+const createWizardMap = (potentialWizardSteps: Steps): WizardMap => {
+  const wizardMap = {};
+  potentialWizardSteps.map((potentialWizardStep, index) => {
```

Comment:
why those steps are potential ? :P

Reply:
I can't remember :)

I think it has something to do with the fact that you can initialize, in theory, the hook with an empty array `[]`, in which case there isn't any step.